### ISSUE TITLE
Support for adding tags to HEIC format images

### ISSUE THREAD

**Issue by jonathan2384** [deckerst/aves#145]
**Is your feature request related to a problem? Please describe.**
It's not strictly a problem, but images taken in the HEIC format are fairly common, especially amongst Apple and Samsung devices. HEIC images, while not being as widely supported as older image formats such as JPEG and PNG, offers many benefits, which include but are not limited to smaller file sizes and better image quality.

**Describe the solution you'd like**
I'd sincerely appreciate a feature to add tags and metadata to images in the HEIC format.

**Describe alternatives you've considered**
Aves is honestly the only app in it's class - the default Samsung Gallery features a "Tags" feature, but does not actually modify the image's metadata itself, causing data loss when switching devices or platforms.

**Additional context**

**Comment by deckerst** [comment]
I agree this is important and I'd like to support this, but the format is not well supported by the libraries that Aves uses to read and write metadata. I'll try and implement it myself someday, but it's unlikely to be in the short term.

**Comment by RohsCE** [comment]
Bumping this, HEIC support would still be quite useful

**Comment by pktiuk** [comment]
@deckerst 

> but the format is not well supported by the libraries that Aves uses to read and write metadata   

Did anything change since 2021?   
Is there at least any bug in in that library's repository which we could track?

-------------------------------------------

It seems that there are already some tools which can work with EXIF in HEIC
https://exiftool.org/

**Comment by deckerst** [comment]
@pktiuk cf https://github.com/deckerst/aves/wiki/Review:-Android-&-Flutter-packages#exif

In Aves, I use a mix of `ExifInterface` and `pixymeta` (both forked by me to fix some bugs and prevent some crashes) to write image metadata. `pixymeta` is unlikely to support new formats. `ExifInterface` could, but I wouldn't hold my breath.

`AndroidX HeifWriter` has an interesting [`addExifData` method](https://developer.android.com/reference/kotlin/androidx/heifwriter/HeifWriter#addExifData(int,byte[],int,int)). But for XMP I'm not sure whether it should be part of Exif or put in a separate segment. I know that `ExifInterface`, which can read HEIF metadata, favours XMP from a separate segment over embedding in Exif tag 700.

`libheif` mentions reading support for Exif/XMP, but not writing. It's unclear if it does.

edit:
- HEIF metadata is described in Annex A of ISO-IEC 23008-12
- HEIF derives from ISOBMFF, which is described in ISO-IEC 14496-12
